prow/test/integration/test/sub_test.go

```diff
@@ -80,6 +129,90 @@ git commit -m "commit 1"
 			},
 			expected: `hello from trigger-postsubmit-via-pubsub1
 this-is-from-repo1
+`,
+		},
+		{
+			name: "inrepoconfig-presubmit",
+			repoSetups: []fakegitserver.RepoSetup{
+				{
+					Name:      "repo2",
+					Script:    CreateRepo2,
+					Overwrite: true,
+				},
+			},
+			msg: fakepubsub.PubSubMessageForSub{
+				Attributes: map[string]string{
+					subscriber.ProwEventType: subscriber.PresubmitProwJobEvent,
+				},
+				Data: subscriber.ProwJobEvent{
+					Name: "trigger-inrepoconfig-presubmit-via-pubsub",
+					Refs: &prowjobv1.Refs{
+						Org:      "org1",
+						Repo:     "repo2",
+						BaseSHA:  Repo2HEADsha,
+						BaseRef:  "master",
+						CloneURI: "http://fakegitserver.default/repo/repo2",
+						Pulls: []prowjobv1.Pull{
+							{
+								Number: 1,
+								SHA:    Repo2PR1sha,
+							},
+							{
+								Number: 2,
+								SHA:    Repo2PR2sha,
+							},
+							{
+								Number: 3,
+								SHA:    Repo2PR3sha,
+							},
+						},
+					},
+				},
+			},
+			expected: `hello from trigger-inrepoconfig-presubmit-via-pubsub
+this-is-from-repo2
+`,
+		},
+		{
+			name: "inrepoconfig-presubmit-with-nested-directory",
+			repoSetups: []fakegitserver.RepoSetup{
+				{
+					Name:      "org1/repo2",
+					Script:    CreateRepo2,
+					Overwrite: true,
+				},
+			},
+			msg: fakepubsub.PubSubMessageForSub{
+				Attributes: map[string]string{
+					subscriber.ProwEventType: subscriber.PresubmitProwJobEvent,
+				},
+				Data: subscriber.ProwJobEvent{
+					Name: "trigger-inrepoconfig-presubmit-via-pubsub",
+					Refs: &prowjobv1.Refs{
+						Org:      "some-prefix/org1",
```

Comment:
can you replace this with `Org: "http://fakegitserver.default/repo/org1",`? I think sub should be smart enough to clone `http://fakegitserver.default/repo/org1/repos`

Reply:
Once #26574 is merged, this should pass. Would recommend add another test case as I mentioned above, iirc sub knows to concatenate `<ORG>/<REPO>` for cloning, so that even if there is no `CloneURI` it would also work.

Reply:
I think you're right. Let me update this PR after I confirm that the integration tests still pass locally.

Reply:
I think there's a little more subtlety involved because of the special meaning of `https://` WRT Gerrit-style org names. We've used `http://` so far in these tests, which is not the same and is not what we expect in production anyway. So I'm going to try to make the tests use https style Org names and see how that goes.

Reply:
I've created https://github.com/kubernetes/test-infra/pull/26587 to clean this part up a little bit, so that we can use "https://..." style refs everywhere in our integration tests. Once we merge that I can update these tests.

Reply:
Done(?). PTAL